src/model/ModelValidation.js

```diff
@@ -3,6 +3,15 @@ import Logger from '../logger/Logger';
 import Api from '../api/Api';
 import { getOwnedPropertyJSON } from './helpers/json';
 
+function extractValidationViolations(webmessage) {
+    if (webmessage.response && webmessage.response.errorReports) {
+        return webmessage.response.errorReports;
```

Comment:
Is this a rejected promise already? If not, then this should be wrapped inside a newly created, rejected promise as well, otherwise the return value types will differ.

Reply:
Yeah it is, that's actually why I'm returning a Promise in the other case as well. Let me see if I can find where this promise is created..

Reply:
Actually I was wrong, it's not a Promise, it's an object:

```javascript
[ { message: 'Required property missing.', property: 'name' } ]
```

Wrapping it in a `Promise.resolve` doesn't break tests, but I'll check it some more. See what's going on there.

Reply:
So, after looking at it, this whole method seems pretty unintuitive. To address your original question here, we don't necessarily have to return the same types as `then` will wrap the return value right? Correct me if I'm wrong. However, as to the weird things I found:

- [This line for example](https://github.com/dhis2/d2/blob/4d9475869a15c82738f4d4c1105ea8e625b63752/src/model/ModelValidation.js#L53) seems pretty pointless to me.
- And [here](https://github.com/dhis2/d2/blob/4d9475869a15c82738f4d4c1105ea8e625b63752/src/model/ModelValidation.js#L58) webmessage is passed to `.reject`, where I'd usually expect an error.
- Also [here](https://github.com/dhis2/d2/blob/4d9475869a15c82738f4d4c1105ea8e625b63752/src/model/ModelValidation.js#L6) the error is caught and resolves succesfully when there are error reports, seems counterintuitive, but I'm not sure what's going on here exactly. 

Seems like the control flow is a bit messed up here. But I think this is something we should address in a separate PR. I'll make some changes and ping you when I've pushed them.